### ISSUE TITLE
another mod seems to be hijacking the shutdown message - also no visible starting msg

### ISSUE THREAD

**Issue by CruxCraft** [nwesterhausen/valheim-discordconnector#18]
We aren't getting the configured shutdown messages at all, and we believe this is the culprit:
HackShardGaming-WorldofValheim-ServerSideCharacters
It announces the server is being shutdown in 10 seconds (via a shout) and we would rather have neither than have that shout show up in discord.

Filtering that message and others out would be amazing, but even then we don't have shutdown messages (we also do not have startup messages, only when it's finished starting)

Hope that is helpful in some way... Looking forward to player death messages and other events =] Thanks!

**Comment by nwesterhausen** [comment]
Interesting interaction. I will have to look into that mod to see how its doing things. I may be able to filter out shouts from "invalid" players but I need to look into what its doing.

**Comment by CruxCraft** [comment]
I am going through the log to see If I can figure anything out, and stumbled upon this:

[Info   :   BepInEx] Loading [Valheim Discord Connector 0.6.0]
[Info   :Valheim Discord Connector] Not running on a dedicated server, some features may break -- please report them!

This is very much a dedicated server... not running from client

I will continue looking through the log

**Comment by nwesterhausen** [comment]
Yep! That is fixed in a patch I'm about to release, 0.7.0

**Comment by CruxCraft** [comment]
this may or may not be helpful, but here this is. I don't see anything really in the log, but maybe something in there will be helpful with a little insight =]


[Info   :   BepInEx] Loading [Valheim Discord Connector 0.6.0]
[Info   :Valheim Discord Connector] Not running on a dedicated server, some features may break -- please report them!
[Info   :Valheim Discord Connector] Flushed cached stats to /home/container/records.json



[Info   : Unity Log] 10/05/2021 19:09:40: Game - OnApplicationQuit
[Info   :   Console] WoV - SSC: Shutdown Detected.
[Info   :   Console] WoV - SSC: Broadcasting Server is being shutdown in 10 seconds by Remote Command! Please disconnect your client now.!!
[Info   :Valheim Discord Connector] Flushed cached stats to /home/container/records.json
[Info   : Unity Log] [Better Trader] Trader's coins were saved with a value of 2459
[Info   : Unity Log] 10/05/2021 19:09:42: Steam manager on destroy

**Comment by nwesterhausen** [comment]
Can you share the records.json? That may have some clues as "who" is doing the shouting, or at least what they list their user name as. 

But I think ultimately I'll have to add some extra logging and try running both mods together. I believe I found the public repository for World of Valheim so I have that to look through as well.

**Comment by CruxCraft** [comment]
[{"Category":"death","Values":[]},{"Category":"join","Values":[{"PlayerName":"\u00C6lfhere","Value":1}]},{"Category":"leave","Values":[]},{"Category":"ping","Values":[]},{"Category":"shout","Values":[{"PlayerName":"WoV - SSC","Value":1},{"PlayerName":"\u00C6lfhere","Value":1},{"PlayerName":"SkipSleep","Value":1}]}]

I see those plugins as players in there... Maybe there is a way to mute a specific player

**Comment by nwesterhausen** [comment]
Yeah it would be easy to add a filter out for player names, to not send a message if it matched one of the list

**Comment by nwesterhausen** [comment]
Going to be able to test my changes after I run an errand but I made the changes to respect the list and add it to config. Attached is the plugin if you want to try it before I get chance.

[DiscordConnector-0.7.1.zip](https://github.com/nwesterhausen/valheim-discordconnector/files/7288926/DiscordConnector.zip)

**Comment by CruxCraft** [comment]
Ignored Players = SkipSleep;WoV - SSC

THIS WORKED =]
at least with blocking WoV - SSC shutdown message... I am going to sleep to test the other, but assume it did too

**Comment by CruxCraft** [comment]
Well... I thought it was just fixed, when in fact shouts are all blocked. It might be blocking those specifically too lol, but it is also blocking my shouts. Were map pings working beforehand too? I didn't enable them until this version, but they arent right now...

`## Valheim Discord Connector v0.7.1
## Plugin GUID: games.nwest.valheim.discordconnector
[Discord Settings]
Webhook URL = [redacted]]
Use fancier discord messages = false
Ignored Players = SkipSleep;WoV - SSC
[Notification Content Settings]
Server Launch Message = The world Könnunsvefnð is awakening!;The world Könnunsvefnð is rising from its slumber!
Server Started Message = Könnunsvefnð is awaiting warriors!;Könnunsvefnð is ready for warriors!
Server Stop Message = The world Könnunsvefnð has gone to rest!;Könnunsvefnð is drifting away!
Player Join Message = has joined Könnunsvefnð!;has entered Könnunsvefnð!;has awakened in Könnunsvefnð!;has arrived in Könnunsvefnð!
Player Leave Message = has left;has moved on.;returns to dreams.
[Notification Settings]
Server Launch Notifications = true
Server Loaded Notifications = true
Server Status Notifications = true
Chat Messages Notifications = true
Chat Shout Messages Notifications = true
Chat Shout Messages Position Notifications = false
Ping Notifications = true
Ping Notificiations Include Position = true
Player Join Notifications = true
Include POS With Player Join = false
Player Leave Notifications = false
Include POS With Player Leave = false
Periodic Player Stats Notifications = true
Player Stats Notifications Period = 4560
[Statistics Collection Settings and Opt-Outs]
Collect Player Stats = true
Collect and Send Player Death Stats = true
Collect and Send Player Join Stats = true
Collect and Send Player Leave Stats = true
Collect and Send Player Ping Stats = true
Collect and Send Player Shout Stats = true
`

**Comment by nwesterhausen** [comment]
Yeah they were working before, clearly my filtering method did not work. I'll be able to tinker with it a bit this evening.

**Comment by CruxCraft** [comment]
> Yeah they were working before, clearly my filtering method did not work. I'll be able to tinker with it a bit this evening.
> 
> Pings are part of the chat messages and its in that area I put the filter in. It must just be extra zealous.

That explains why pings still didnt show up when I rolled back to 0.7.0... I let the other plugin handle the shouts for now. I like the way yours writes the shouts better though, so looking forward to it just handing death messages, as my webhook is more of a persona and I didnt like their name getting changed for the messages...

thanks for the quick effort earlier, even if it didnt succeed. at the very least I realized I didnt have to try using regex in the other and could just block users... still helped ;]

**Comment by nwesterhausen** [comment]
By "extra join when player dies" you mean the join message after the death message?

Will fix the coords added to all message, I think a couple don't add them.

Periodic leaderboard not yet enabled in the code

**Comment by nwesterhausen** [comment]
Leaderboard working for upcoming 0.9.0

![image](https://user-images.githubusercontent.com/2317381/136225664-8248e8fc-db4f-4940-82a5-42a69fae4eac.png)

**Comment by CruxCraft** [comment]
I wrote UNTEST as I leave those disabled and was only going to test them if I ran out of things to do. We only use the coords for pings, as we are somewhat PVP... We havent decided if we even want leave messages yet as we have mixed thoughts on using them, but I am learning towards it...

Ohhh lol... I didnt realize leaderboards werent enabled yet lol... I was just curious to see it... I was going to wait to install th
e newer version as I have some other projects, but now I am intrigued and will do it after breakfast =]

I'll go through and test everything again sometime today too...

**Comment by ekloswick** [comment]
It's on Windows, running on my own computer. I do have other mods - I can test later tonight if disabling them still has the issue, though I'm definitely thinking it's getting shut down instantly now that you mention it. It doesn't do a save when I CTRL+C the cmd window, where it looks like yours does.

This is the .sh file I'm running to launch the server (just modified the exec call), and it supposedly supports CTRL+C.

```
#!/bin/sh
# BepInEx-specific settings
# NOTE: Do not edit unless you know what you are doing!
####
export DOORSTOP_ENABLE=TRUE
export DOORSTOP_INVOKE_DLL_PATH=./BepInEx/core/BepInEx.Preloader.dll
export DOORSTOP_CORLIB_OVERRIDE_PATH=./unstripped_corlib

export LD_LIBRARY_PATH="./doorstop_libs:$LD_LIBRARY_PATH"
export LD_PRELOAD="libdoorstop_x64.so:$LD_PRELOAD"
####

export LD_LIBRARY_PATH="./linux64:$LD_LIBRARY_PATH"
export SteamAppId=892970

echo "Starting server PRESS CTRL-C to exit"

# Tip: Make a local copy of this script to avoid it being overwritten by steam.
# NOTE: Minimum password length is 5 characters & Password cant be in the server name.
# NOTE: You need to make sure the ports 2456-2458 is being forwarded to your server through your local router & firewall.
#
exec "<folderPath>\steamapps\common\Valheim dedicated server/valheim_server.exe" -nographics -batchmode -name "<serverName>" -port 2457 -world "<worldName>" -password "<password>" -public 1
```

**Comment by nwesterhausen** [comment]
Here is a version of 1.2.1 where this mod patches itself before WoV-SSC in the shutdown prefix. This theoretically fixes the compatibility issue but need confirmation.

[DiscordConnector-WoV-SSC-compatTest.zip](https://github.com/nwesterhausen/valheim-discordconnector/files/7365830/DiscordConnector-WoV-SSC-compatTest.zip)
